### PR TITLE
[GeckoView] Add a Nimbus experiment for the toolbar (bug 1833093)

### PR DESCRIPTION
**Pull request by calixteman**

### CODE REVIEW

File: web/app.js [hunk 1]
```diff
@@ -152,6 +152,10 @@ class DefaultExternalServices {
   static get canvasMaxAreaInBytes() {
     return shadow(this, "canvasMaxAreaInBytes", -1);
   }
+
+  static getNimbusExperimentData() {
+    return shadow(this, "getNimbusExperimentData", Promise.resolve(null));
+  }
```
Comment (Snuffleupagus) [inline web/app.js:158]:
```suggestion
  static async getNimbusExperimentData() {
    return shadow(this, "getNimbusExperimentData", null);
  }
```

**Review by Snuffleupagus — COMMENTED** [review]
Given that I don't know what "Nimbus" is, it's sort of difficult to properly review that part of the patch :-)

Also, given that this essentially looks like another way to control the visibility of the GeckoView-toolbar, could we perhaps remove the `enableFloatingToolbar`-pref in favor of this new code?

**Comment by calixteman** [comment]
Nimbus is a framework to make some experiments:
https://experimenter.info/

Some people will have one button or two or none.

**Comment by calixteman** [comment]
> Given that I don't know what "Nimbus" is, it's sort of difficult to properly review that part of the patch :-)
> 
> Also, given that this essentially looks like another way to control the visibility of the GeckoView-toolbar, could we perhaps remove the `enableFloatingToolbar`-pref in favor of this new code?

We'll remove this pref asap (likely in this cycle) but here Nimbus is not a pref system at all but a way to do some experiments to try to understand what users want.

File: web/app.js [hunk 2]
```diff
@@ -587,7 +593,12 @@ const PDFViewerApplication = {
           : PDFJSDev.test("GECKOVIEW")
       ) {
         if (AppOptions.get("enableFloatingToolbar")) {
-          this.toolbar = new Toolbar(appConfig.toolbar, eventBus, this.l10n);
+          this.toolbar = new Toolbar(
+            appConfig.toolbar,
+            eventBus,
+            this.l10n,
+            this.nimbusData
```
[the remark this answers is not in the document]
Comment (calixteman) [reply web/app.js]: I think that the values are retrieved from a server, hence it's probably better to call `getNimbus...` as soon as possible (without awaiting) and then await when it's needed.
So I changed the code in order to store a Promise and only await on it at the toolbar creation.

**Review by Snuffleupagus — APPROVED** [review]

**Pull request merged**